internal/project/service.go

```diff
@@ -787,3 +795,13 @@ func (s *Service) printProjects() {
 func (s *Service) logf(format string, args ...any) {
 	s.Log(fmt.Sprintf(format, args...))
 }
+
+// !!! per root compiler options
+func (s *Service) SetCompilerOptionsForInferredProjects(compilerOptions *core.CompilerOptions) {
+	s.compilerOptionsForInferredProjects = compilerOptions
```

Comment:
This is probably going to require some form of synchronization when we do start calling it.

Just to expand further on what I think we should do as an alternative:

The LSP allows servers to call back into the client and ask for configuration (e.g., VS Code settings), and also ask the client to send a notification when that configuration has changed. This means that if we want to allow the user to override the default configuration, we'd call back into the client for the info and register a "watcher" on it. Then, any configuration changes would come in as plain messages and be correctly synchronized by the handler loop.

This is not something I would expect in this PR whatsoever, obviously.